### PR TITLE
fixes #3404 : refactored susi-chatbot.js to ES6 and eliminate the use of jQuery

### LINKED ISSUE #3404: refactor susi-chatbot.js
**Expected Behaviour**

refactor susi-chatbot.js to ES6 and eliminate the use of jQuery.

**Proposed IN**
https://github.com/fossasia/susi.ai/pull/3397#pullrequestreview-351573920

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by anishagg17**
Fixes #3404

Changes:  eliminated the use of jQuery

Demo Link: https://pr-3407-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

File: public/susi-chatbot.js [hunk 1]
```diff
@@ -35,125 +35,132 @@ let botbuilderBotMessageTextColor = '#455a64';
...
+  ] = botbuilderBackgroundBody;
+
   if (botbuilderBodyBackgroundImg) {
-    $('.susi-sheet-content-container').css(
-      'background-image',
-      "url('" + botbuilderBodyBackgroundImg + "')",
-    );
+    getComputedStyle(document.querySelector('.susi-sheet-content-container'))[
+      'background-image'
+    ] = "url('" + botbuilderBodyBackgroundImg + "')";
```
Comment (Dilshaad21) [inline public/susi-chatbot.js]: I think we can use ES6 template literals here

File: public/susi-chatbot.js [hunk 2]
```diff
@@ -35,125 +35,132 @@ let botbuilderBotMessageTextColor = '#455a64';
...
-      'background-image',
-      "url('" + botbuilderIconImg + "')",
-    );
+    getComputedStyle(document.querySelector('.susi-launcher-button'))[
+      'background-image'
+    ] = "url('" + botbuilderIconImg + "')";
+
+    getComputedStyle(document.querySelector('.susi-comment-avatar'))[
+      'background-image'
+    ] = "url('" + botbuilderIconImg + "')";
```
Comment (Dilshaad21) [inline public/susi-chatbot.js]: and here

File: public/susi-chatbot.js (hunk 1 above)
Comment (anishagg17) [reply public/susi-chatbot.js]: it's not necessary the soul purpose is to remove jQuery

File: public/susi-chatbot.js (hunk 2 above)
Comment (kpulkit29) [reply public/susi-chatbot.js]: @anishagg17 Template literals would be better and easier to understand.

**Comment by anishagg17** [comment]
@kpulkit29 @Dilshaad21 now i have used template strings

File: public/susi-chatbot.js [hunk 3]
```diff
@@ -33,223 +33,203 @@ let botbuilderUserMessageTextColor = '#ffffff';
...
 
 function enableBot() {
-  $(document).ready(function() {
+  let ready = callback => {
+    if (document.readyState != 'loading') callback();
+    else document.addEventListener('DOMContentLoaded', callback);
+  };
+
+  ready(() => {
     const baseUrl = api_url + '/susi/chat.json?q=';
```
Comment (Dilshaad21) [inline public/susi-chatbot.js]: Changes not made here, else looks good.

Comment (anishagg17) [reply public/susi-chatbot.js]: I didn't saw that 
Thanks for pointing 👍

**Comment by anishagg17** [comment]
@Dilshaad21 please checkout for more such errors

File: public/susi-chatbot.js [hunk 4]
```diff
@@ -33,223 +33,203 @@ let botbuilderUserMessageTextColor = '#ffffff';
...
-    'https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js';
-  jqTag.onload = getTheme;
-  headTag.appendChild(jqTag);
-} else {
-  // to ensure jquery has loaded
-  getTheme();
-}
+let botbuilderIconImg = `${susi_skills_deployed_url}customAvatars/0.png`;
+
+getTheme();
```
Comment (akshatnitd) [inline public/susi-chatbot.js]: Use IIFE here. (https://medium.com/javascript-in-plain-english/https-medium-com-javascript-in-plain-english-stop-feeling-iffy-about-using-an-iife-7b0292aba174)

File: public/susi-chatbot.js [hunk 5]
```diff
@@ -416,46 +381,34 @@ function enableBot() {
...
-        '<div class="susi-comment-body ">' +
-        '<div class="susi-comment-content">' +
-        val +
-        '</div>' +
-        '</div>' +
-        '</div>' +
-        '</div>' +
-        '</div>';
-      $(UserResponse).appendTo('.susi-conversation-parts');
+      const UserResponse = `<div class="susi-conversation-part susi-conversation-part-grouped-first"><div class=" susi-comment susi-comment-by-user "><div class="susi-comment-body-container susi-comment-body-container-user" style="background-color:${botbuilderUserMessageBackground};color:${botbuilderUserMessageTextColor}"><div class="susi-comment-body "><div class="susi-comment-content">${val}</div></div></div></div></div>`;
```
Comment (akshatnitd) [inline public/susi-chatbot.js]: `userResponse`

File: public/susi-chatbot.js [hunk 6]
```diff
@@ -35,125 +35,132 @@ let botbuilderBotMessageTextColor = '#455a64';
...
-          ? settings.botIconImage
-          : botbuilderIconImg;
-        applyTheme();
-      }
-    },
-    error: function(e) {
-      console.log(e);
-    },
-  });
+  script = document.createElement('script');
```
[the remark this answers is not in the document]
Comment (anishagg17) [reply public/susi-chatbot.js]: okay

[the remark this answers is not in the document]
Comment (anishagg17) [reply public/susi-chatbot.js]: Sir , that script is necessary I had searched a lot to accept jsonp datatype using fetch .
This was the only way to do so .

[the remark this answers is not in the document]
Comment (akshatnitd) [reply public/susi-chatbot.js]: @anishagg17 Is `jsonp` required here? Try without `jsonp`, it should work.

**Comment by anishagg17** [comment]
@akshatnitd  changes have been done

**Pull request merged**